autoload/db_ui/schemas.vim

```diff
@@ -85,11 +85,27 @@ let s:mysql = {
       \ 'layout_flag': '\\G',
       \ 'quote': 0,
       \ }
+
+let s:oracle_args = "echo \"SET linesize 4000;\nSET pagesize 4000;\nCOLUMN owner FORMAT a20;\nCOLUMN table_name FORMAT a25;\nCOLUMN column_name FORMAT a25;\n%s\" | "
+let s:oracle = {
+\   'args': s:oracle_args,
+\   'cell_line_number': 1,
+\   'cell_line_pattern': '^-\+\( \+-\+\)*',
+\   'default_scheme': '',
+\   'foreign_key_query': printf(s:oracle_args, "SELECT DISTINCT RFRD.table_name, RFRD.column_name, RFRD.owner FROM all_cons_columns RFRD JOIN all_constraints CON ON RFRD.constraint_name = CON.r_constraint_name JOIN all_cons_columns RFRING ON CON.constraint_name = RFRING.constraint_name JOIN all_users U ON CON.owner = U.username WHERE CON.constraint_type = 'R' AND U.common = 'NO' AND RFRING.column_name = '{col_name}';"),
```

Comment:
Lets extract these big queries into their own variables, like we did for other db types.

Reply:
The latest commit aims to address this.

I also took some steps to abstract `sqlplus`'s strict query rules by joining lists with delimiters. Hopefully that will make maintaining Oracle support easier, since formatting the expressions happens in one place.